packages/forms/src/UIForm/merge.js

```diff
@@ -0,0 +1,207 @@
+import React, { PropTypes } from 'react';
+import { merge } from 'talend-json-schema-form-core';
+import get from 'lodash/get';
+
+import FieldTemplate from './fields/FieldTemplate';
+
+/* eslint-disable no-param-reassign */
+
+function parseArray(items, key) {
+	if (items.type === 'object') {
+		return parseProperties(items.properties, false, `${key}[]`);
+	}
+	return [getUISchemaFromObject(items, key)];
+}
+
+function getUISchemaFromObject(schema, key) {
+	const ui = { key };
+	if (schema.title) {
+		ui.title = schema.title;
+	}
+	if (schema.description) {
+		ui.description = schema.description;
+	}
+	if (schema.enum && schema.enumNames) {
+		ui.titleMap = schema.enum.map((value, index) => ({ value, name: schema.enumNames[index] }));
+		delete schema.enumNames;
+	}
+	if (schema.type === 'object') {
+		ui.widget = 'fieldset';
+		ui.items = parseProperties(schema.properties, false, key);
+	} else if (schema.type === 'string') {
+		if (!schema.enum) {
+			ui.widget = 'text';
```

Comment:
I think this is not necessary, by default, widget is `text`. And if we have an enum in jsonSchema, by default it is a select.
You can see that in action in `core-text.js` and `core-select.js`